src/main/java/dog/pawbook/logic/parser/PawbookParser.java

```diff
@@ -72,35 +74,47 @@ public Command parseCommand(String userInput) throws ParseException {
         default:
             throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
         }
-
-
     }
 
-    private AddCommand generateAddCommand(String entityType, String arguments) throws ParseException {
+    /**
+     * Generate add command based on the given user input.
+     *
+     * @param entityType entityType from user input
+     * @param arguments arguments from user input
+     * @return an add command based on the user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddCommand generateAddCommand(String entityType, String arguments) throws ParseException {
         if (entityType.isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
-
-        switch (entityType) {
-        case AddCommand.ENTITY_WORD:
-            return new AddCommandParser().parse(arguments);
-
-        default:
-            throw new ParseException(MESSGAE_UNKNOWN_ENTITY);
+        if (arguments.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+        }
+        if (!AddCommand.ENTITY_WORD.equals(entityType)) {
+            throw new ParseException(MESSAGE_UNKNOWN_ENTITY);
         }
+        return new AddCommandParser().parse(arguments);
     }
 
-    private DeleteCommand generateDeleteCommand(String entityType, String arguments) throws ParseException {
+    /**
+     * Generate delete command based on the given user input.
+     *
+     * @param entityType entityType from user input
+     * @param arguments arguments from user input
+     * @return a delete command based on the user input
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteCommand generateDeleteCommand(String entityType, String arguments) throws ParseException {
```

Comment:
I like that you have added in the javadocs, however, this is to be kept private as it is an internal helper method that should not be exposed to any other users. Perhaps you made it public in order to write test code for it, but I think it is better to refrain from that since it is the public API that we are testing out, the test code should not care about the actual implementation so long as it does what it promises. Else we will fall into this never ending issue of making all methods of every class public. I could have inlined these two lines into the switch case of the main `parse` method, but for SLAP I created the helper functions.

Reply:
Yea, I made it public just for the testing. Does that mean I only have to test public methods that are written?

Reply:
Yes, you can take a look at the rest of the code base and see how it is done for the others. Especially for helper methods that are more concerned with internal implementation, they could be very dynamic and change every time.